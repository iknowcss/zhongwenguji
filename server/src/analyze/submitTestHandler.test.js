const express = require('express');
const bodyParser = require('body-parser');
const supertest = require('supertest');
const submitTestHandler = require('./submitTestHandler');

describe('submitTestHandler', () => {
  let body;

  beforeAll(() => {
    const app = express();
    app.post('/', bodyParser.json(), submitTestHandler);
    return supertest(app)
      .post('/')
      .send({
        testData: [
          { isTested: true, knownPercent: 100, range: [0, 250] },
          { isTested: false, knownPercent: NaN, range: [250, 500] },
          { isTested: true, knownPercent: 80, range: [500, 750] },
          { isTested: true, knownPercent: 40, range: [750, 1000] },
          { isTested: true, knownPercent: 60, range: [1000, 1250] },
          { isTested: true, knownPercent: 20, range: [1250, 1500] },
          { isTested: true, knownPercent: 0, range: [1500, 1750] },
          { isTested: true, knownPercent: 0, range: [1750, 2000] },
          { isTested: false, knownPercent: NaN, range: [2000, 2250] },
          { isTested: false, knownPercent: NaN, range: [2250, 2500] }
        ]
      })
      .expect(200)
      .then((result) => {
        body = result.body;
      });
  });

  it('calculates curve points', () => {
    const { curvePoints } = body;
    expect(curvePoints.slice(0, 5).map(([x]) => x))
      .toEqual([0, 249, 498, 747, 996]);
  });

  it('calculates sample points', () => {
    const { samplePoints } = body;
    expect(samplePoints).toEqual([
      [125, 100],
      [623, 80],
      [872, 40],
      [1121, 60],
      [1370, 20],
      [1619, 0],
      [1868, 0]
    ]);
  });

  it('known character range estimate', () => {
    const { knownEstimate, knownEstimateUncertainty } = body;
    expect({ knownEstimate, knownEstimateUncertainty }).toMatchSnapshot();
  });
});
