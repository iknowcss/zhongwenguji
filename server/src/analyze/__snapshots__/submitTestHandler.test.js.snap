// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitTestHandler known character range estimate 1`] = `
Object {
  "knownEstimate": 1000,
  "knownEstimateUncertainty": 100,
}
`;
